doc/nrf/releases_and_maturity/migration/migration_3.1_54h_suit_ironside.rst

```diff
@@ -0,0 +1,177 @@
+
+* :kconfig:option:`CONFIG_SUIT`
+* :kconfig:option:`CONFIG_SUIT_SECURE_DOMAIN`
+
+Enable MCUboot for device firmware update (DFU) in the :file:`sysbuild.conf` file as follows:
+
+* Set :kconfig:option:`SB_CONFIG_BOOTLOADER_MCUBOOT` to ``y``.
+
+
+Update devicetree files
+-----------------------
+
+You must update your devicetree files as follows:
+
+* Remove the old UICR partition.
+  In your board's DTS overlay, remove any node that defined the ``uicr`` partition.
+
+* Add the PERIPHCONF array.
+  In your devicetree, under the ``mram1x`` partitions node, define a partition node labeled ``peripconf_partition`` with a size of at least 8 KB to embed the generated address-value blob.
+
+* Remove IPC-shared-memory reservation.
+  As IronSide relocates the IPC buffer to a fixed RAM20 address, you can delete any manual reservation in RAM0.
+  Refer to the `Memory map changes`_ section.
+
+* Update IPC configuration for IronSide SE.
+  The shared memory for communication with the Secure Domain now uses fixed addresses in ``RAM20``.
+  A single memory region is used for both RX and TX operations.
+  The IPC nodes use the ``nordic,ironside-call`` compatible and communicate using the new *IronSide Calls* IPC driver.
+
+  For custom board devicetree files, you can copy the IPC configuration from the nRF54H20 DK reference implementation.
+  The devicetree defines the shared memory region and IPC nodes as follows:
+
+  .. code-block:: dts
+
+     // Shared memory region in RAM20
+     cpusec_cpuapp_ipc: memory@2f88f000 {
+         reg = <0x2f88f000 DT_SIZE_K(4)>;
+     };
+
+  .. code-block:: dts
+
+     // IPC nodes using IronSide calls driver
+     cpusec_cpuapp_ipc_tx: ipc@deadbeef {
+         compatible = "nordic,ironside-call";
+         mboxes = <&cpuapp_cpusec_ipc 0>, <&cpuapp_cpusec_ipc 1>;
+         mbox-names = "rx", "tx";
+         memory-region = <&cpusec_cpuapp_ipc>;
+         status = "okay";
+     };
+
+  .. code-block:: dts
+
+     cpusec_cpuapp_ipc_rx: ipc@deadbeef {
+         compatible = "nordic,ironside-call";
+         mboxes = <&cpusec_cpuapp_ipc 2>, <&cpuapp_cpusec_ipc 3>;
+         mbox-names = "rx", "tx";
+         memory-region = <&cpusec_cpuapp_ipc>;
+         status = "okay";
+     };
+
+Update PERIPHCONF
+-----------------
+
+The new UICR format no longer holds peripheral configuration initial values.
+You must generate a PERIPHCONF blob at build time.
+
+Update PERIPHCONF
+-----------------
+
+The new UICR format no longer holds peripheral configuration initial values.
+You must generate a PERIPHCONF blob at build time.
+
+The Zephyr build invokes the :file:`gen_uicr.py` script (:file:`soc/nordic/common/uicr/gen_uicr.py` in the Zephyr tree) via ``nrf-regtool`` in the NCS sysbuild system.
+When the following Kconfig options are set:
+
+  * :kconfig:option:`CONFIG_NRF_HALTIUM_GENERATE_UICR` to ``y``
+  * :kconfig:option:`CONFIG_NRF_HALTIUM_UICR_PERIPHCONF` to ``y``
+
+The script does the following:
```

Comment:
Unordered list used instead of steps. "The following" suggests ordered actions.

Reply:
Aligning with the criteria followed in the previous chapter.